### PR TITLE
Add connection name validation

### PR DESCRIPTION
**Pull request by ptrf**
This patch solves the connection name validation in #39

If accepted, this PR makes turtle fail on attempts to start publishers or services that reference a connection that has not been defined in the turtle connection_config, as described under the Configuration section in the README

### CODE REVIEW

**Comment by ptrf** [comment]
The failing travis-ci build is for Erlang 19.3. There's an issue with the ranch_proxy_protocol dependency, which fails to unpack. I've previously run into this issue, which has to do with `erl_tar` in 19.3 failing with `{error, bad_header}` error, when attempting to unpack the ranch_proxy_protocol package.

I have yet to report this issue.

**Comment by jlouis** [comment]
There is a `Changes` section in the `README.md`. Could you also add to that one so we can track the changes we are doing to the system?

**Comment by ptrf** [comment]
Thanks! Will do

File: README.md [hunk 1]
```diff
@@ -92,6 +92,17 @@ become worse.
 
 # Changes
 
+* *Version 1.9.4* - Introduce Connection Name Validation
```
Comment (jlouis) [inline README.md]: Perhaps just "unreleased" since we have yet to push it out.

Comment (ptrf) [reply README.md]: Well, I cut a new release, since the 1.9.3 didn't build with Erlang 19.3

**Pull request merged**